src/main/scala/org/allenai/pipeline/Pipeline.scala

```diff
@@ -181,6 +215,38 @@ trait ConfiguredPipeline extends Pipeline {
 
     logger.info(s"Summary written to $outputDir")
   }
+}
+
+trait ConfiguredPipeline extends Pipeline {
+  def config: Config
+
+  private[this] val persistedStepsByConfigKey =
+    scala.collection.mutable.Map.empty[String, Producer[_]]
+
+  private lazy val runOnlySteps = config.get[String]("runOnly").map(_.split(",").toSet).getOrElse(Set.empty[String])
```

Comment:
Lazy is evil, as in it's a source of gotchas and lengthy debug sessions. Perhaps lazy + Spark is much worse than usual lazy, but I've come to be very suspicious around lazy.

Do we really need lazy here? If we do, can we please add a blurb explaining why?